Sources/HiPMobile/DataAccessLayer/IJoinEntity.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.Shared.DataAccessLayer
```

Comment:
I find this whole file difficult to understand. Any chance you could document its usage with a few examples on Confluence or in a GitHub wiki page and link it here?

Reply:
I added some more documentation and linked to the blog article I have taken most of the code from. I think the article explains the problem and the idea very well.